crates/extensions/c8y_mapper_ext/src/operations/config_update.rs

```diff
@@ -237,92 +137,41 @@ impl CumulocityConverter {
             cmd_id: cmd_id.to_string(),
         };
         let topic = self.mqtt_schema.topic_for(&target.topic_id, &channel);
-        let external_id: String = target.external_id.clone().into();
 
-        // Replace '/' with ':' to avoid creating unexpected directories in file transfer repo
-        let tedge_url = format!(
-            "http://{}/tedge/file-transfer/{}/config_update/{}-{}",
-            &self.config.tedge_http_host,
-            external_id,
-            smartrest.config_type.replace('/', ":"),
-            cmd_id
-        );
+        let proxy_url = self
+            .c8y_endpoint
+            .maybe_tenant_url(&smartrest.url)
+            .map(|cumulocity_url| self.auth_proxy.proxy_url(cumulocity_url).into());
+
+        let remote_url = proxy_url.unwrap_or(smartrest.url.to_string());
```

Comment:
I would raise a discussion for later. This change will mix up the meaning of `remoteUrl` field.

In config_update, the `remoteUrl` is used only to download a file. So proxy url as `remoteUrl` seems fine.
In firmware_update, the `remoteUrl` is necessary to but it has to be the original url (no proxy url!). Otherwise, in the end, c8y mapper cannot report the current firmware including url.

I think we need two fields, one indicates the original request URL, and the other indicates the url where actually `tedge-agent` downloads the file.